### PR TITLE
Smooth Quad Distortion

### LINKED ISSUE #4288: Smooth Quad Distortion
#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Other (specify if possible)

#### New feature details:
Quad distortion now:
https://editor.p5js.org/Vamoss/sketches/vWdWD1q4
image

Smooth squad demo suggestion:
https://editor.p5js.org/Vamoss/sketches/1iaogZo2
image

### PR DESCRIPTION
**Pull request by Vamoss**
Resolves #4288

Changes:
Add  detailX and detailY parameters to quad function, so users can smooth the quad distortion, it is very visible with textures as described in this issue #4288

#### PR Checklist
- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

### CODE REVIEW

**Comment by stalgiag** [comment]
Thank you @Vamoss !

The WebGL side looks great but I'm not sure I understand the purpose of adding the detailX and detailY for the 2DRenderer. What was your thinking there?

**Comment by lmccart** [comment]
@Vamoss can you give us some insight into the question of detailX and detailY from @stalgiag above?

**Comment by Vamoss** [comment]
Hi @stalgiag and @lmccart, super sorry for te delay.

I think @stalgiag is right, there is no poimlnt to have it in 2D mode.
The only advantage of this chance is for the webgl methods.

I can make a new pull request in the next few days to restore the original 2D code.

**Comment by Vamoss** [comment]
Hi,

I just restored the `p5.Renderer2D.js` file.

Event dow I was not sure, I kept the changes in `2d_primitives.js`:
https://github.com/processing/p5.js/pull/4901/commits/75b8afdfe7f03f7bec395904b137c7f1b8d940f3#diff-9a3dbc4b3ce17b1abf41b9910db0d839b30179f1c8864f0de0d5ba42e1d73d9dL522

I hope everything is right, let me know if I can do anything else :)

Cheers,
Carlos

File: src/core/shape/2d_primitives.js [hunk 1]
```diff
@@ -488,6 +488,8 @@ p5.prototype.point = function(...args) {
  * @param {Number} y3 the y-coordinate of the third point
  * @param {Number} x4 the x-coordinate of the fourth point
  * @param {Number} y4 the y-coordinate of the fourth point
+ * @param {Integer} [detailX] number of segments in the x-direction
+ * @param {Integer} [detailY] number of segments in the y-direction
```
Comment (lmccart) [inline src/core/shape/2d_primitives.js:492]: can you remove these two lines?

File: src/core/shape/2d_primitives.js [hunk 2]
```diff
@@ -513,21 +515,24 @@ p5.prototype.point = function(...args) {
  * @param {Number} x4
  * @param {Number} y4
  * @param {Number} z4 the z-coordinate of the fourth point
+ * @param {Integer} [detailX]
+ * @param {Integer} [detailY]
```
Comment (lmccart) [inline src/core/shape/2d_primitives.js:519]: can you remove these two lines?

**Comment by lmccart** [comment]
thanks @Vamoss this looks great! @stalgiag do you want to merge this one if it looks all good to you?

**Comment by stalgiag** [comment]
Thank you @Vamoss

**Pull request merged**